tiledb/sm/metadata/metadata.cc

```diff
@@ -283,10 +283,10 @@ void Metadata::swap(Metadata* metadata) {
   std::swap(loaded_metadata_uris_, metadata->loaded_metadata_uris_);
 }
 
-void Metadata::reset() {
+void Metadata::reset(uint64_t timestamp) {
   clear();
-  auto t = utils::time::timestamp_now_ms();
-  timestamp_range_ = std::make_pair(t, t);
+  timestamp = (timestamp != 0) ? timestamp : utils::time::timestamp_now_ms();
```

Comment:
is 0 a valid `open_at` value?

Reply:
Yes, so that the previous code works. `0` practically means "now". The writer picks that up as well and produces a fragment with the current timestamp (e.g., in consolidation).